flash/image/classification/model.py

```diff
@@ -87,59 +89,58 @@ def __init__(
         learning_rate: float = 1e-3,
         multi_label: bool = False,
         serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+        training_strategy: Optional[str] = "default",
+        training_strategy_kwargs: Optional[Dict[str, Any]] = None,
     ):
-        super().__init__(
-            num_classes=num_classes,
-            model=None,
-            loss_fn=loss_fn,
-            optimizer=optimizer,
-            optimizer_kwargs=optimizer_kwargs,
-            scheduler=scheduler,
-            scheduler_kwargs=scheduler_kwargs,
-            metrics=metrics,
-            learning_rate=learning_rate,
-            multi_label=multi_label,
-            serializer=serializer or Labels(multi_label=multi_label),
-        )
 
         self.save_hyperparameters()
 
         if not backbone_kwargs:
             backbone_kwargs = {}
 
+        if not training_strategy_kwargs:
+            training_strategy_kwargs = {}
+
+        if training_strategy_kwargs != "default":
+            if "ways" in training_strategy_kwargs and training_strategy_kwargs["ways"] != num_classes:
+                raise MisconfigurationException(
+                    "When providing ways, it should match `num_classes` as mapping is not supported yet."
```

Comment:
```suggestion
                    "When providing `ways` in `training_strategy_kwargs`, it should match `num_classes` as mapping is not supported yet. Specify `num_classes` to your desired value of `ways`."
```
```suggestion
                    "When providing ways, it should match `num_classes` as mapping is not supported yet."
```

Reply:
The proposal above is a minor revision.

A somewhat more general revision would be to always overwrite `num_classes` with `ways` and warn the user about it. What do you think?

Even more general would be to revisit the `num_classes` argument, as discussed with @ethanwharris. However, this is a larger discussion and I think it does not fit in this PR. About this, though, I'd be curious to know your (on-liner) opinion :)